content/docs/rfcs/17/README.md

```diff
@@ -2,127 +2,138 @@
 
-For the registration, a peer creates a transaction that sends x (TODO to be specified) ETH to the contract. 
-The peer who has the "private key" `sk` associated with that deposit would be able to withdraw x ETH by providing valid proof. 
-Note that  `sk` is initially only known by the owning peer however it may get exposed to other peers in case the owner attempts spamming the system i.e., sending more than one message per epoch.
 
-<!-- TODO: the interaction with the contract is subject to change depending on the final implementation -->
 
-<!-- Once registered, the peer obtains the root of the tree (after the registration of the current peer) i.e., `root`  as well as the authenticity path `authPath`. A peer can prove her membership using the `authPath`. -->
+## Setup and Registration
+Peers subscribed to a specific `pubsubTopic` form a [RLN group](/spec/32).
+<!-- link to the RLN group definition in the RLN RFC -->
+Peers MUST be registered to the RLN group to be able to publish messages.
+Registration is moderated through a smart contract deployed on the Ethereum blockchain. 
+Each peer has an [RLN key pair](/spec/32) denoted by `sk`  and `pk`.
+The  secret key `sk` is secret data and MUST be persisted securely by the peer.
+The state of the membership contract contains the list of registered members' public identity keys i.e., `pk`s. 
+For the registration, a peer creates a transaction that invokes the registration function of the contract via which registers its `pk` in the group. 
+The transaction also transfers some amount of ether to the contract to be staked. 
+This amount is denoted by `Staked_Fund` and is a system parameter.
+The peer who has the secret key `sk` associated with a registered `pk` would be able to withdraw a portion `Reward_Portion` of the staked fund by providing valid proof. <!-- a secure way to prove the possession of a pk is yet under discussion, maybe via commit and reveal -->
+`Reward_Portion` is also a system parameter.
+
+Note that  `sk` is initially only known to its owning peer however, it may get exposed to other peers in case the owner attempts spamming the system i.e., sending more than one message per `epoch`.
+An overview of registration is illustrated in Figure 1.
 
+![Figure 1: Registration.](../../../../rfcs/17/rln-relay.png)
 
-<!-- `sk`  and `authPath`  are secret data and MUST be permanently and locally stored by the peer.  -->
 
-<!-- TODO: To specify the details of protobuf messages for the interaction with the contract -->
+## Publishing
 
-<!-- diagram -->
+To publish at a given `epoch`, the publishing peer proceeds based on the regular [`11/WAKU2-RELAY`](/spec/11) protocol.  
+However, to protect against spamming, each `WakuMessage` (which is wrapped inside the `data` field of a PubSub message) MUST carry a [`RateLimitProof`](##RateLimitProof) with the following fields. 
```

Comment:
```suggestion
To publish at a given `epoch`, the publishing peer proceeds based on the regular [`11/WAKU2-RELAY`](/spec/11) protocol.

However, to protect against spamming, each `WakuMessage` (which is wrapped inside the `data` field of a PubSub message) MUST carry a [`RateLimitProof`](##RateLimitProof) with the following fields. 
```

Reply:
Like this, same across document. Makes it hard to read in plain text mode.

Reply:
doen in https://github.com/vacp2p/rfc/pull/484/commits/8954039fe162421351bcacf73cba41ef864bca21